lib/mix/lib/mix/dep/lock.ex

```diff
@@ -62,4 +63,11 @@ defmodule Mix.Dep.Lock do
   defp lockfile do
     Mix.Project.config[:lockfile]
   end
+
+  defp assert_no_merge_conflicts_in_lockfile(lockfile, info) do
+    if info =~ ~r/^(?:<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|)/ do
```

Comment:
Ok, maybe we can go with `~r/^<<<<<<</m` then? Git conflicts should all have this marker.